optuna_dashboard/_app.py

```diff
@@ -269,17 +269,29 @@ def save_study_note(study_id: int) -> dict[str, Any]:
     @json_api_view
     def post_preference(study_id: int) -> dict[str, Any]:
         try:
-            best_trials = [int(d) for d in request.json.get("best_trials", [])]
-            worst_trials = [int(d) for d in request.json.get("worst_trials", [])]
-        except ValueError:
+            mode = request.json.get("mode", "")
+            candidates = [int(d) for d in request.json.get("candidates", [])]
+            clicked = int(request.json.get("clicked", -1))
+        except Exception:
             response.status = 400
-            return {"reason": "best_trials and worst_trials must be an array of integers."}
-        if len(best_trials) == 0 or len(worst_trials) == 0:
-            response.status = 400  # Bad request
-            return {"reason": "You need to set best_trials and worst_trials"}
+            return {"reason": "Invalid request."}
 
-        preferences = [(best, worst) for best in best_trials for worst in worst_trials]
-        report_preferences(study_id, storage, preferences)
+        if clicked == -1:
```

Comment:
nit: Could you improve the error handling so we can see what exactly went wrong?